file_scanning/main.go

```diff
@@ -0,0 +1,144 @@
+package main
```

Comment:
Other than the other comments most the code looks good to me. Only nit-picks I have are more about how we should follow the Go's handbook called [Effective Go](https://go.dev/doc/effective_go) when writing comments etc. I'll point them out in the review but will leave a `(can be ignored)` on the comment

Reply:
Thanks Ste for the comments. They were really useful. I've added all the changes suggested. It was good doing the reading on the go standards. All really good stuff :)